lib/puppet/parser/functions/new.rb

```diff
@@ -864,7 +866,7 @@
 | b | The data is in base64 encoding, padding as required by base64 strict is added by default
 | u | The data is in URL safe base64 encoding
 | B | The data is in base64 strict encoding
-| s | The data is a puppet string used verbatim as a binary byte sequence
+| s | The data is a puppet string seen as a binary sequence of bytes with ignored encoding
```

Comment:
The specification can state that for Ruby, the string will retain its encoding and used byte by byte. Other (future) platforms must create text with the correct encoding such that when mapped to a String, and then to a Binary using Binary.new(str, '%s') that it retains the read original byte sequence verbatim. It can also state that functions should not do this, and instead directly return Binary.